data/org.freedesktop.portal.DynamicLauncher.xml

```diff
@@ -0,0 +1,223 @@
+      <arg type="s" name="parent_window" direction="in"/>
+      <arg type="s" name="name" direction="in"/>
+      <arg type="h" name="icon_fd" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="handle" direction="out"/>
+    </method>
+    <!--
+        RequestInstallToken:
+        @name: The name that will be used in the desktop file
+        @icon_fd: An open file descriptor for the icon file
+        @options: Vardict with optional further information
+        @token: the token to be used with the Install() method, or the empty
+          string if there was an error
+
+        This method is intended for use only by trusted system components and
+        will fail if used by a sandboxed process. The token returned by this
+        method can be used to avoid the need for a confirmation dialog; the
+        token is passed to the Install() method just as if it were acquired
+        via the PrepareInstall() method.
+
+        The @options vardict currently has no supported entries.
+    -->
+    <method name="RequestInstallToken">
+      <annotation name="org.gtk.GDBus.C.UnixFD" value="true"/>
+      <arg type="s" name="name" direction="in"/>
+      <arg type="h" name="icon_fd" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="s" name="token" direction="out"/>
+    </method>
+    <!--
+        Uninstall:
+        @desktop_file_id: The ID of a desktop file
+        @options: Vardict with optional further information
+        @result: 0 to indicate success, 1 if the app was already not installed,
+          some other number to indicate failure
+
+        This method deletes the desktop file and corresponding icon from the
+        appropriate directories to remove the launcher referred to by
+        @desktop_file_id.
+        If the calling process is sandboxed, the @desktop_file_id must be
+        prefixed by the app ID followed by a period. For example, Epiphany might
+        use an ID like
+        "org.gnome.Epiphany.WebApp_e9d0e1e4b0a10856aa3b38d9eb4375de4070d043.desktop"
+        In that example the desktop file would exist at the path
+        "~/.local/share/xdg-desktop-portal/applications/org.gnome.Epiphany.WebApp_e9d0e1e4b0a10856aa3b38d9eb4375de4070d043.desktop"
+        with a sym link in "~/.local/share/applications/".
+        The checksum at the end of the file name is an implementation detail in
+        Epiphany and not required by the portal.
+
+        If the calling process is unsandboxed, @desktop_file_id need only have
+        a ".desktop" suffix.
+
+        This method is intended to be called by the application that created the
+        launcher, e.g. Epiphany, so it can clean up associated data such as the
+        profile directory as part of the uninstallation. The proper way for a
+        software center to remove an Epiphany-installed web app is using the
+        org.gnome.Epiphany.WebAppProvider D-Bus interface.
```

Comment:
This reads a bit too GNOMEy to me. Portals are DE-agnostic, so any mention to DEs must be clearly only as examples. Allow me to suggest the following:

```
        This method is intended to be called by the application that created the
        launcher, e.g. a web browser, so it can clean up associated data such as
        the profile directory as part of the uninstallation. Consequently, the
        proper way for a software center to remove a launcher is by using the APIs
        provided by the application that installed it. For example, for GNOME
        Software to remove web launchers created by Epiphany, it would use the
        org.gnome.Epiphany.WebAppProvider D-Bus interface.
```